token-metadata/js/src/instruction.ts

```diff
@@ -0,0 +1,56 @@
+import type { PublicKey } from '@solana/web3.js';
+
+import { Field, TokenMetadata } from './state';
+
+
+/*
+Not implimented as a strict interface, by function signatures so that the 
+library can work with programs that implement the Rust interface.
+*/
+
+
+// TODO:- Library should impliment Pack/unpack helpers
+export type Pack = (meta: TokenMetadata) => Promise<Buffer>
+export type Unpack = (input: Buffer) => Promise<TokenMetadata>
+
+export type Initialize = (
+    programId: PublicKey,
+    metadata: PublicKey,
+    updateAuthority: PublicKey,
```

Comment:
this could also be optional

Reply:
Just want to confirm my understanding/implementation on the optionality of the updateAuthority.

For all instructions updateAuthority is not optional. It could be optional for `updateAuthority`, but i think its better to have users explicitly set the new Authority explicitly as the null address.

When retrieving the TokenMetadata - updateAuthority might the be null address, for these cases i have implimented such that it shows up as undefined, rather than the default/null public key.

Reply:
Actually after checking out the instruction it's apparently not optional when you initialize, so we're good here. It's only optional when you go to update it!